### PR TITLE
Feature ag grid tables

### PR DESCRIPTION
**Pull request by Austral1a**

### CODE REVIEW

File: src/containers/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -1,36 +1,71 @@
...
 const {
-  tabsItems: { tab1, tab2, tab3 },
+  tabsItems: { tab1, tab2, tab3, tab4 },
 } = translation;
 
 const tabValues: TabValues[] = [
   {
     title: <Link to={TabRoute.tab1}>{tab1.title}</Link>,
-    content: <TabRoutes routePath={TabRoute.tab1} component={LazyTab1} />,
+    content: (
```
Comment (MagroTanya) [inline src/containers/Tabs/Tabs.tsx:29]: why did you add round brackets?

File: src/containers/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -1,36 +1,71 @@
...
+  getBAEmployeesAction(),
+  getPMEmployeesAction(),
+  getProgrammersEmployeesAction(),
+];
+
 export const Tabs: FC = (): ReactElement => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    getEmployeesActions.forEach((getEmployeeAction) => {
```
Comment (MagroTanya) [inline src/containers/Tabs/Tabs.tsx:64]: move dispatch to each list

File: src/helpers/routes.ts [hunk 3]
```diff
@@ -10,4 +10,5 @@ export enum TabRoute {
   tab1 = "/Tabs/tab/1",
```
Comment (MagroTanya) [inline src/helpers/routes.ts:10]: change urls according to tabs names

File: src/components/common/Tab/hooks/useTabManager.tsx [hunk 4]
```diff
@@ -19,16 +19,28 @@ export const useTabManager = (
...
-    width: tabWidth,
-    left: tabWidth * (activeTab + 1) - tabWidth,
-  };
+  const [activeLinePosition, setActiveLinePosition] = useState<number>(0);
 
   useEffect(() => {
-    setActiveTab(activeTab);
+    setActiveLinePosition(
+      Object.values(tabHeaderRef.current.children)
+        // collect only heading elements
```
Comment (MagroTanya) [inline src/components/common/Tab/hooks/useTabManager.tsx:27]: move logic to the method with an appropriate name

File: src/containers/Tabs/Tabs.tsx (hunk 1 above)
Comment (Austral1a) [reply src/containers/Tabs/Tabs.tsx:29]: I didn't, "Prettier" did - for readability purposes I guess...

File: src/containers/Tabs/Tabs.tsx (hunk 2 above)
Comment (Austral1a) [reply src/containers/Tabs/Tabs.tsx:64]: I will, but if I'd like to change dispatch method I'll need to change it as many times as I have actions rather than once - in "forEach" loop.

File: src/components/common/Tab/hooks/useTabManager.tsx [hunk 5]
```diff
@@ -19,16 +19,38 @@ export const useTabManager = (
 
   const [tabWidth, setTabWidth] = useState<number>(0);
 
-  const activeLineStyle = {
-    width: tabWidth,
-    left: tabWidth * (activeTab + 1) - tabWidth,
-  };
+  const [activeLinePosition, setActiveLinePosition] = useState<number>(0);
+
+  const sumTabsWidthTilActiveTab = useCallback(() => {
```
Comment (MagroTanya) [inline src/components/common/Tab/hooks/useTabManager.tsx:24]: better name calculateActiveLinePosition

File: src/helpers/routes.ts [hunk 6]
```diff
@@ -6,8 +6,9 @@ export enum RoutePath {
   settings = "/settings",
 }
 
-export enum TabRoute {
-  tab1 = "/Tabs/tab/1",
-  tab2 = "/Tabs/tab/2",
-  tab3 = "/Tabs/tab/3",
-}
+export const tabRoutes = {
+  businessAnalysts: `${RoutePath.tabs}/businessAnalysts/tab/1`,
```
Comment (MagroTanya) [inline src/helpers/routes.ts:10]: what is the purpose of using / tab / 1? better use `${RoutePath.tabs}/businessAnalysts`

Comment (Austral1a) [reply src/helpers/routes.ts:10]: In order to track the opened tab, after page reload.

File: src/components/common/Tab/components/TabItem/TabItem.tsx [hunk 7]
```diff
@@ -5,17 +5,23 @@ import classes from "./TabItem.module.scss";
...
   onClick,
   children,
+  dataTabName,
 }): ReactElement => {
   const tabItemClasses = classNames(classes["tab__title"], className);
 
   return (
-    <h2 onClick={onClick} className={tabItemClasses}>
+    <h2
+      data-tab-name={dataTabName}
```
Comment (MagroTanya) [inline src/components/common/Tab/components/TabItem/TabItem.tsx:21]: why not camelcase?
